File: src/main/java/org/example/eventhub/dto/event/EventFilter.java
package org.example.eventhub.dto.event;

import org.example.eventhub.enums.EventStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record EventFilter(
        String title,
        String city,

        Integer minCapacity,
        Integer maxCapacity,

        BigDecimal minPrice,
        BigDecimal maxPrice,

        EventStatus eventStatus,

        LocalDateTime fromDateTime,
        LocalDateTime toDateTime,
        Boolean upcoming
) {
}

File: src/main/java/org/example/eventhub/dto/event/EventFilter.kt
package org.example.eventhub.dto.event;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import org.example.eventhub.enums.EventStatus;

public record EventFilter(
        String title,
        String city,
        Integer minCapacity,
        Integer maxCapacity,
        BigDecimal minPrice,
        BigDecimal maxPrice,
        EventStatus eventStatus,
        LocalDateTime fromDateTime,
        LocalDateTime toDateTime,
        Boolean upcoming) {}
